src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -272,6 +270,7 @@ bool BulletRigidObject::removeObject() {
   }
   bWorld_.reset();  // release shared ownership of the world
   rigidObjectType_ = RigidObjectType::NONE;
+  delete &node();
```

Comment:
After node() is deleted, 'this' no longer exists. That's confusing. This function seems like it's the destructor. Why can't teardown be implemented with unique_ptrs and destructors instead of calls to removeObject from the BulletPhysicsManager destructor?

Reply:
Yeah, the semantics of `removeObject()` is a bit confusing now ... since this is a feature, isn't it enough to remove the feature and keep the object as-is (so it loses its rigid body capabilities)?

Reply:
Good point, I've refactored the deconstruction/removal process.